### DISCUSSION TITLE
Error

### DISCUSSION THREAD

**Discussion by Marklord-dev** [Cyfrin/foundry-full-course-cu#2052] [Q&A]
I'm getting an installation verification error on my foundry-zksync installation. Please what am i not doing right?
![forge error](https://github.com/user-attachments/assets/f21afb0e-da25-439e-9dd3-5b8de370209c)

**Comment by EngrPips** [comment]
I am unsure what is wrong, but maybe uninstall Foundry then re-install it using this `https://book.getfoundry.sh/getting-started/installation` Foundry installation guide.

**Reply by Marklord-dev** [reply]
Did that already and still facing the same problem.

**Reply by EngrPips** [reply]
This `https://github.com/matter-labs/foundry-zksync` is the official repo of foundry-zkync. I think you should follow the **Quick Install** section of the repo, and you should successfully install `foundry-zksync`

**Reply by Marklord-dev** [reply]
i followed all the procedure on the quick install. i don't know why it couldn't install and verify forge

**Reply by Marklord-dev** [reply]
i was able to install foundry but foundry-zksync keeps giving me issues

**Reply by sivaren** [reply]
@Marklord-dev It's probably because the system requires `GLIBC 2.32` version or higher. 

We can update the library by running commands below:
```
sudo apt update
sudo apt install libc6
```

Afterward, check its current version:
```
ldd --version
```

If the GLIBC version requirement is already satisfied, then it worked well. But **if it's not,** you need to consider upgrading the Ubuntu/WSL version with the **LTS version.** 

Then check the distribution version and your GLIBC library to ensure your current status:
```
lsb_release -a
ldd --version
```

Lastly, reinstall your `foundyup` or `foundryup-zksync` and I hope it can solve the problem🤞

**Reply by EngrPips** [reply]
Interesting.

**Comment by Marklord-dev** [comment]
forge --version keeps returning this 

forge --version 
forge: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by forge)
forge: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by forge)
forge: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by forge)

**Reply by EngrPips** [reply]
I don't know what is wrong as everything works for me smoothly. Installation always seems to be the most challenging part of programming. Let's hope some Giga Chad see this and help. @cromewar, your help is highly needed, sir.

**Comment by SimSimButDifferent** [comment]
I had the same issue, managed to fix the "GLIBC_2.32' not found" by updating ubuntu. I hadn't updated it on WSL in 2 years...

Here are the steps:

First update WSL:

```
wsl.exe --update
``` 

Then check for updates for good measure:

```
sudo apt update && sudo apt upgrade
```

Then upgrade ubuntu to the latest version:

```
sudo do-release-upgrade
```

Once its done close all terminal instances of wsl.

After that I re-installed foundry-zksync and followed the usual install steps and everything worked fine.

**Reply by EngrPips** [reply]
Thanks for this sir @SimSimButDifferent, @Marklord-dev  Please try this out and let's see if it solve your issue.

**Reply by marina-ciuperca** [reply]
thank you @SimSimButDifferent for sharing your knowledge! Your instructions worked like a charm for my machine.

**Reply by EngrPips** [reply]
Awesome

**Reply by SimSimButDifferent** [reply]
> thank you @SimSimButDifferent for sharing your knowledge! Your instructions worked like a charm for my machine.

Glad to hear it worked for you

**Comment by soarewee** [comment]
I had the same problem, and updating from Ubuntu 20.04 LTS to 22.04 fixed it for me as well 
Here’s the devcontainer.json I used to set up Ubuntu 22.04 in GitHub Codespaces:

```json
{
    "name": "Ubuntu 22.04",
    "image": "mcr.microsoft.com/devcontainers/base:ubuntu-22.04",
    "features": {
        "ghcr.io/devcontainers/features/common-utils:2": {
            "installZsh": "true",
            "username": "vscode",
            "upgradePackages": "true"
        }
    },
    "remoteUser": "vscode"
}
```

**Comment by EngrPips** [comment]
Interesting.

**Comment by TatyOko28** [comment]
You're missing the required GLIBC versions (2.32, 2.33, 2.34) needed to run Foundry. You need to either:

Update GLIBC: Install the required versions using your package manager:

For Ubuntu/Debian:
```shell
sudo apt update && sudo apt install libc6
```
For Arch:
```shell
sudo pacman -S glibc
```
For Fedora:
```shell
sudo dnf install glibc
```
Use a prebuilt GLIBC version:
Download and compile a newer version of GLIBC:
```shell
wget http://ftp.gnu.org/gnu/libc/glibc-2.34.tar.gz
tar -xvzf glibc-2.34.tar.gz
cd glibc-2.34
mkdir build
cd build
../configure --prefix=/opt/glibc-2.34
make -j$(nproc)
sudo make install
```
Then, run Foundry using:
```shell
LD_LIBRARY_PATH=/opt/glibc-2.34/lib ./foundry
```

**Reply by EngrPips** [reply]
Interesting.
